gc/base/standard/Scavenger.cpp

```diff
@@ -784,15 +779,52 @@ MM_Scavenger::mergeGCStats(MM_EnvironmentBase *env)
 		scavStats->_releaseScanListCount);
 }
 
+void
+MM_Scavenger::mergeIncrementGCStats(MM_EnvironmentBase *env, bool lastIncrement)
+{
+	Assert_MM_true(env->isMasterThread());
+	MM_ScavengerStats *finalGCStats = &_extensions->scavengerStats;
+	mergeGCStatsBase(env, finalGCStats, &_extensions->incrementScavengerStats);
+
+	/* Language specific stats were supposed to be merged directly from thread local to cycle global. No need to merge them here. */
+
+	if (lastIncrement) {
+		finalGCStats->getFlipHistory(0)->_tenureMask = _tenureMask;
+		uintptr_t tenureAge = 0;
+		for (tenureAge = 0; tenureAge <= OBJECT_HEADER_AGE_MAX; ++tenureAge) {
```

Comment:
Can you put a brief explanation for this code in a comment?

Reply:
I don't know exactly what it does. It was just moved out from common merge path, into the marge path that is done only once by master thread. Back there it was executed by every GC thread and protected with is-master-thread check.

Reply:
There seem to be two parts, one is "Recalculate tenure age" and the other "Update Flip/tenure stats for age 0". Makes sense to you ?

Reply:
Sorry I missed that it was removed from the other location.  I thought it was new code.  Since it is not new code it is ok as is.

Reply:
I added those two comments anyway